libraries/node-core-library/src/FileSystem.ts

```diff
@@ -785,6 +791,59 @@ export class FileSystem {
     });
   }
 
+  /**
+   * Writes the contents of multiple Uint8Arrays to a file on disk, overwriting the file if it already exists.
+   * Behind the scenes it uses `fs.writevSync()`.
+   * This API is useful for writing large files efficiently, especially if the input is being concatenated from
+   * multiple sources.
+   * @remarks
+   * Throws an error if the folder doesn't exist, unless ensureFolder=true.
+   * @param filePath - The absolute or relative path of the file.
+   * @param contents - The content that should be written to the file.
+   * @param options - Optional settings that can change the behavior.
+   */
+  public static writeBuffersToFile(
+    filePath: string,
+    contents: Iterable<Uint8Array>,
+    options?: IFileSystemWriteBinaryFileOptions
+  ): void {
+    FileSystem._wrapException(() => {
+      const toCopy: Uint8Array[] = [...contents];
+
+      let fd: number | undefined;
+      try {
+        fd = fsx.openSync(filePath, 'w');
```

Comment:
(Interesting that we never exposed options for appending (`a`) etc.)

Reply:
We have a completely separate API for append.